src/openrct2/scripting/bindings/entity/ScGuest.hpp

```diff
@@ -16,6 +16,66 @@
 
 namespace OpenRCT2::Scripting
 {
+    static const DukEnumMap<ShopItem> ShopItemMap({
+        { "beef_noodles", ShopItem::BeefNoodles },
+        { "burger", ShopItem::Burger },
+        { "candyfloss", ShopItem::Candyfloss },
+        { "chicken", ShopItem::Chicken },
+        { "chips", ShopItem::Chips },
+        { "chocolate", ShopItem::Chocolate },
+        { "cookie", ShopItem::Cookie },
+        { "doughnut", ShopItem::Doughnut },
+        { "hot_dog", ShopItem::HotDog },
+        { "fried_rice_noodles", ShopItem::FriedRiceNoodles },
+        { "funnel_cake", ShopItem::FunnelCake },
+        { "ice_cream", ShopItem::IceCream },
+        { "meatball_soup", ShopItem::MeatballSoup },
+        { "pizza", ShopItem::Pizza },
+        { "popcorn", ShopItem::Popcorn },
+        { "pretzel", ShopItem::Pretzel },
+        { "roast_sausage", ShopItem::RoastSausage },
+        { "sub_sandwich", ShopItem::SubSandwich },
+        { "tentacle", ShopItem::Tentacle },
+        { "toffee_apple", ShopItem::ToffeeApple },
+        { "wonton_soup", ShopItem::WontonSoup },
+        { "coffee", ShopItem::Coffee },
+        { "drink", ShopItem::Drink },
+        { "fruit_juice", ShopItem::FruitJuice },
+        { "iced_tea", ShopItem::IcedTea },
+        { "lemonade", ShopItem::Lemonade },
+        { "soybean_milk", ShopItem::SoybeanMilk },
+        { "sujeonggwa", ShopItem::Sujeonggwa },
+        { "balloon", ShopItem::Balloon },
+        { "hat", ShopItem::Hat },
+        { "map", ShopItem::Map },
+        { "sunglasses", ShopItem::Sunglasses },
+        { "toy", ShopItem::Toy },
+        { "tshirt", ShopItem::TShirt },
+        { "umbrella", ShopItem::Umbrella },
+        { "photo1", ShopItem::Photo },
+        { "photo2", ShopItem::Photo2 },
+        { "photo3", ShopItem::Photo3 },
+        { "photo4", ShopItem::Photo4 },
+        { "voucher", ShopItem::Voucher },
+        { "empty_bottle", ShopItem::EmptyBottle },
+        { "empty_bowl_blue", ShopItem::EmptyBowlBlue },
+        { "empty_bowl_red", ShopItem::EmptyBowlRed },
+        { "empty_box", ShopItem::EmptyBox },
+        { "empty_burger_box", ShopItem::EmptyBurgerBox },
+        { "empty_can", ShopItem::EmptyCan },
+        { "empty_cup", ShopItem::EmptyCup },
+        { "empty_drink_carton", ShopItem::EmptyDrinkCarton },
+        { "empty_juice_cup", ShopItem::EmptyJuiceCup },
+        { "rubbish", ShopItem::Rubbish },
+    });
```

Comment:
Can you `static_assert` (or assert) that the ShopItemMap has the same size as the enum `ShopItem`? That would make it easier to know where to update if we add/remove items.

PS: Somehow `ShopItem::Count` doesn't reflect the actual number of ShopItems, but it's alright to hardcode an operation for now, the point is for it to fail if it changes from here onwards.

Reply:
I'll admit I'm not entirely sure how to get the size of `ShopItemMap`. There isn't a property for a size, and the `end()` property seems to just be an iterator. I could theoretically iterate through the list to increment a size variable, though that seems like a less than ideal solution.

Also when you mention to hardcode a value, does that mean something like `static_assert(ShopItemMapSize == (ShopItem.Count - 6);` since that's the amount of skipped entries (and one for `ShopItem::Admission` which guests cannot carry, as it's used for restrooms to sell and whatnot)? I'm not entirely sure what would be most helpful to detect a change in the size of `ShopItem`.

Reply:
> Also when you mention to hardcode a value, does that mean something like static_assert(ShopItemMapSize == (ShopItem.Count - 6); since that's the amount of skipped entries (and one for ShopItem::Admission which guests cannot carry, as it's used for restrooms to sell and whatnot)?

Yes, at least we would get some warning somewhere that this is out of sync.

Reply:
What you came up with was exactly what I was going for, it is just a way to immediately get feedback that something will go wrong if the `ShopItem` increases or decreases.

Thanks!